### PR TITLE
Add serial/legacy tablet backend

### PR DESCRIPTION
**Pull request by adryzz**
- [x] Add device streams and endpoints
- [ ] Figure out configs
- [x] Figure out UI
- [ ] Figure out the extra bugs with this
> - Figure out auto-detection of tablet another day in a separate PR
- [ ] Figure out report parser
- [ ] Useless extra dopamine releasing checkbox

### CODE REVIEW

File: OpenTabletDriver/Devices/SerialBackend/SerialInterface.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        public int ProductID { get; private set; }
+
+        public int VendorID { get; private set; }
+
+        public int InputReportLength { get; private set; }
+
+        public int OutputReportLength { get; private set; }
+
+        public int FeatureReportLength => 0; // requires parsing report descriptor to determine feature report length
```
Comment (Sublimelime) [inline OpenTabletDriver/Devices/SerialBackend/SerialInterface.cs:24]: Convert to xmldoc instead of inline comment

**Review by Sublimelime — CHANGES_REQUESTED** [review]

File: OpenTabletDriver.UX/Windows/Tablet/LegacyTabletConnectionWindow.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            
+            reconnectBox = new CheckBox
+            {
+                Text = "Remember tablet"
+            };
+
+            Content = new StackLayout
+            {
+                Padding = 5,
+                Spacing = 5,
```
[the remark this answers is not in the document]
Comment (adryzz) [reply OpenTabletDriver.UX/Windows/Tablet/LegacyTabletConnectionWindow.cs:55]: they are exactly like this in other classes
for example this
https://github.com/OpenTabletDriver/OpenTabletDriver/blob/4f06b708ccc7109e39bd029fcb47f0df3efe992a/OpenTabletDriver.UX/Windows/DeviceStringReader.cs#L65

**Comment by InfinityGhost** [comment]
Marking as draft as this is still incomplete

File: OpenTabletDriver.UX/Windows/Tablet/LegacyTabletConnectionWindow.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (InfinityGhost) [reply OpenTabletDriver.UX/Windows/Tablet/LegacyTabletConnectionWindow.cs:55]: It's fine to leave this non-constant.

File: OpenTabletDriver/Devices/SerialBackend/SerialDeviceHub.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+using System.IO.Ports;
+using OpenTabletDriver.Plugin.Attributes;
+using OpenTabletDriver.Plugin.Devices;
+
+namespace OpenTabletDriver.Devices.SerialBackend
+{
+    [DeviceHub]
+    public class SerialDeviceHub : ILegacyDeviceHub
+    {
+        public SerialDevoceHub()
```
Comment (InfinityGhost) [inline OpenTabletDriver/Devices/SerialBackend/SerialDeviceHub.cs:10]: Typo in constructor
```suggestion
        public SerialDeviceHub()
```

**Review by InfinityGhost — CHANGES_REQUESTED** [review]

**Comment by adryzz** [comment]
rebased everything, now it's time to fix all the issues and it should work... took 5 hours to rebase...

**Pull request closed without merging**